spec/Section 6 -- Execution.md

```diff
@@ -143,6 +146,127 @@ ExecuteMutation(mutation, schema, variableValues, initialValue):
+the following capabilities:
+
+  * **Must** support observation of the associated publish stream (for example,
+  via iteration, callbacks, or reactive semantics).
+  * **Must** support cancellation of the observation, see {Unsubscribe}.
+  * **Must** support a way to send data to the subscriber.
+  * **May** include an initial response associated with executing the selection
+  set defined on the subscription operation.
+
+Most subscriptions can only be evaluated when event data is available. For
+example, a subscription tells us when users log on and off would require event
+data that tells us *who* logged on or off. Without this event data, the
+subscription cannot be evaluated. However, it is possible to define
+subscriptions that can be evaluated without event data (for example, the current
+time on the server, synthetically triggered every second). Subscriptions that do
+not require event data for evaluation can optionally return an initial response
+to the Subscribe() operation.
+
+Subscribe(schema, subscription, operationName, variableValues, initialValue):
+
+  * Let {subscriptionType} be the root Subscription type in {schema}.
+  * Assert: {subscriptionType} is an Object type.
+  * Let {selectionSet} be the top level Selection Set in {subscription}.
+  * Let {rootField} be the first top level field in {selectionSet}.
+  * Let {eventStream} be the result of running {MapSubscriptionEvents(rootField, variableValues)}.
+  * Optionally: let {data} be the result of running {ExecuteRequest(schema, document, operationName, variableValues, initialValue)}, and return {data} on the subscription object.
+
+MapSubscriptionEvents(rootField, variableValues):
+
+  * *Application-specific logic to map from root field/variables to events*
+
+#### Publish
+
+Once a subscription is created, we listen for events on its event stream. Each
+event carries an optional payload, which we combine with the arguments from
+Subscribe() to resolve the selection set.
+
+  * For each {event} and {eventData} on {eventStream}:
+    * Let {data} be the result of running
+      {ExecuteSelectionSet(selectionSet, subscriptionType, eventData, variableValues)}
+      *normally* (allowing parallelization).
+    * Let {errors} be any *field errors* produced while executing the
+      selection set.
+    * If {errors} is not empty, optionally, Unsubscribe() the subscription.
+    * Yield an unordered map containing {data} and, optionally, {errors} on
+      {publishStream}.
+  * Let {publishStream} be the sequence of outputs yielded above.
+
+#### Unsubscribe
+
+The unsubscribe operation can be implemented in a number of ways. For example,
+by using a dedicated subscription manager, defining it as a method on the
+subscription object, or cancelling the iterator.
+
+Unsubscribe()
+
+  * Terminate {publishStream} (For example, cancel iteration, detach mapping function)
+  * Terminate and clean up {eventStream}
```

Comment:
This whole section seems like implementation detail and not really relevant to what is or isn't GraphQL, could it be collapsed into a footnote in the top section on subscriptions?